flaml/data.py

```diff
@@ -197,8 +197,21 @@ def fit_transform(self, X, y, task):
             for column in X.columns:
                 # sklearn\utils\validation.py needs int/float values
                 if X[column].dtype.name == 'datetime64[ns]':
```

Comment:
This check should be moved after the check for drop (line 234). Also, the added features should go through the same check for uniqueness before getting added.